system/ee/EllisLab/ExpressionEngine/Model/Member/Member.php

```diff
@@ -342,8 +345,11 @@ public function onAfterUpdate($changed)
 
 		if (isset($changed['password']))
 		{
-			// email the current email address telling them their password changed
-			$this->notifyOfChanges('password_changed_notification', $this->email);
+			if (strlen($changed['password']) == strlen($this->password))
+			{
+				// email the current email address telling them their password changed
+				$this->notifyOfChanges('password_changed_notification', $this->email);
+			}
```

Comment:
I just caught that we have two `if (isset($changed['password']))` blocks. Would you mind combining them into one? One to log the change the other to conditionally email the member?

Reply:
Discussed- it's all good, as the log is inside a cp conditional check.